app/jobs/send_assessments_job.rb

```diff
@@ -5,15 +5,70 @@ class SendAssessmentsJob < ApplicationJob
   queue_as :mailers
 
   def perform(*_args)
-    eligible = Patient.reminder_eligible.count
-    sent = []
+    patient_ids = Patient.reminder_eligible.pluck(:id)
+    eligible = patient_ids.size
+
+    # Check what the max thread pool is for the DB since that is the limiting factor
+    # on how many threads can be used in the job. Subtracting 2 from the pool
+    # size to be on the safer side of exhausting the connection pool.
+    total_threads = [1, ActiveRecord::Base.connection_pool.stat[:size] - 2].max
+
+    # [1, x].max here avoids an exception when there are 0 eligible patients
+    slice_size = [1, patient_ids.size / total_threads].max
+
+    threads = patient_ids.each_slice(slice_size).map do |patient_batch|
+      Thread.new { perform_batch(patient_batch) }
```

Comment:
What was your reasoning for using `Thread` instead of something like `Process.fork` and then `Process.waitall`? (perhaps the real question is about concurrency vs. parallelism and which makes sense here)

Reply:
I can assess the performance of those alternatives on Monday

Reply:
Since we want to bring our data back together at the end to combine and send the `assessment_job_email`, I think it makes more sense to keep threading - for its data sharing and returning friendliness. Forking isn't very friendly to data sharing and returning values (all examples I could find of returning values from a fork felt like hacks).